### PR TITLE
Feature/initial commit

### PR DESCRIPTION
**Pull request by slambang**
Initial commit for review.

### CODE REVIEW

File: dependencies.gradle [hunk 1]
```diff
@@ -0,0 +1,44 @@
+ext {
+    compileSDK = 28
+
+    androidBuildVersion = "28.0.3"
```
Comment (brettwold) [inline dependencies.gradle]: You don't need this (androidBuildVersion) anymore if you're using latest gradle version(s) as the plugin has its own build tools version

Comment (brettwold) [reply dependencies.gradle]: Looks like you're not using it in the build scripts anyway :+1: so you cane remove from here

File: dms-client--simulator-app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    package="com.aevi.sdk.dms.simulator">
+
+    <uses-permission android:name="com.aevi.SESSION_CHANGE"/>
```
Comment (brettwold) [inline dms-client--simulator-app/src/main/AndroidManifest.xml]: This doesn't look right

File: dms-client--simulator-app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    <android.support.constraint.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        tools:context=".ui.SimulatorView">
+
+        <EditText
+            android:id="@+id/uin_field"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="8dp"
```
Comment (brettwold) [inline dms-client--simulator-app/src/main/res/layout/activity_main.xml]: dimens

File: dms-client--simulator-app/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            android:layout_marginTop="8dp"
```
Comment (brettwold) [inline dms-client--simulator-app/src/main/res/layout/activity_main.xml]: dimens

File: dms-client--simulator-app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            android:layout_marginEnd="8dp"
```
Comment (brettwold) [inline dms-client--simulator-app/src/main/res/layout/activity_main.xml]: dimens

Comment (brettwold) [reply dms-client--simulator-app/src/main/res/layout/activity_main.xml]: You get the idea :)

File: dms-client/src/main/java/com/aevi/sdk/dms/DmsClient.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+    public static final String FIELD_UIN = "uin";
+
+    public static final String FIELD_ACCOUNT_ID = "id";
+    public static final String FIELD_ACCOUNT_NAME = "name";
+    public static final String FIELD_ACCOUNT_ROLE = "role";
+
+    private static final Uri SDK_CONTENT_URI = Uri.parse("content://com.aevi.dms.info");
+
+//    private abstract class RxProviderMethod<T> extends Single<T> {
```
Comment (brettwold) [inline dms-client/src/main/java/com/aevi/sdk/dms/DmsClient.java]: If this isn't needed then remove it

File: dms-client/src/main/java/com/aevi/sdk/dms/DmsInfo.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+    private final String uin;
+
+    DmsInfo(@Nullable String uin) {
+        this.uin = uin;
+    }
+
+    /**
+     * @return the unique identifier number of this device. This number isn't attached to a physical device but is
+     * unique amongst the list of active devices. // TODO - Could we explain this as an "installation ID"?
```
Comment (brettwold) [inline dms-client/src/main/java/com/aevi/sdk/dms/DmsInfo.java]: Yes some better explaination might be required here. Maybe ask Eddie for final wording

File: dms-client/src/main/java/com/aevi/sdk/dms/DmsClient.java (hunk 6 above)
Comment (slambang) [reply dms-client/src/main/java/com/aevi/sdk/dms/DmsClient.java]: Done.

File: dms-client--simulator-app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (slambang) [reply dms-client--simulator-app/src/main/AndroidManifest.xml]: Removed.

File: dependencies.gradle (hunk 1 above)
Comment (slambang) [reply dependencies.gradle]: Done.

File: dms-client--simulator-app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (slambang) [reply dms-client--simulator-app/src/main/res/layout/activity_main.xml]: Done.

File: dms-client--simulator-app/src/main/res/layout/activity_main.xml (hunk 4 above)
Comment (slambang) [reply dms-client--simulator-app/src/main/res/layout/activity_main.xml]: Done.

File: dms-client--simulator-app/src/main/res/layout/activity_main.xml (hunk 5 above)
Comment (slambang) [reply dms-client--simulator-app/src/main/res/layout/activity_main.xml]: All done.

File: dms-client/src/main/java/com/aevi/sdk/dms/DmsInfo.java (hunk 7 above)
Comment (slambang) [reply dms-client/src/main/java/com/aevi/sdk/dms/DmsInfo.java]: I've pinged Eddie about this

File: .circleci/config.yml [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    environment:
+      _JAVA_OPTIONS: "-Xms512m -Xmx1024m"
+      GRADLE_OPTS: -Dorg.gradle.daemon=false
+
+reusable-steps:
+  - &restore_cache
+    restore_cache:
+      keys:
+      - v1-dependencies-{{ checksum "build.gradle" }}
+      # fallback to using the latest cache if no exact match is found
```
Comment (louis-barman) [inline .circleci/config.yml]: These two lines should be deleted.
```
      # fallback to using the latest cache if no exact match is found
      - v1-dependencies-
```

File: .circleci/config.yml [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+    working_directory: ~/repo
+    environment:
+      _JAVA_OPTIONS: "-Xms512m -Xmx1024m"
+      GRADLE_OPTS: -Dorg.gradle.daemon=false
+
+reusable-steps:
+  - &restore_cache
+    restore_cache:
+      keys:
+      - v1-dependencies-{{ checksum "build.gradle" }}
```
Comment (louis-barman) [inline .circleci/config.yml]: And all `v1-dependencies` changed to `v2-dependencies`

**Comment by louis-barman** [comment]
It is missing the `writegradleprops.sh` file!

**Comment by louis-barman** [comment]
As discussed with steve you probably need to delete `publish-repo-jar.gradle` and replace with the `android-publish.gradle` see https://github.com/Aevi-UK/android-pos-print-api/blob/develop/android-publish.gradle here for an example

File: build.gradle [hunk 10]
```diff
@@ -0,0 +1,57 @@
+buildscript {
+    repositories {
+        google()
+        jcenter()
+    }
+
+    dependencies {
+        classpath 'com.android.tools.build:gradle:3.2.1'
```
Comment (louis-barman) [inline build.gradle]: Needs these two lines adding here:
```
        classpath 'com.github.dcendents:android-maven-gradle-plugin:2.0'
        classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.8.0'
```

File: .circleci/config.yml (hunk 8 above)
Comment (slambang) [reply .circleci/config.yml]: Done.

File: .circleci/config.yml (hunk 9 above)
Comment (slambang) [reply .circleci/config.yml]: Done.

File: SimulatorApp/build.gradle [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+
+    defaultConfig {
+        applicationId "com.aevi.sdk.dms.enabled"
+        minSdkVersion 21
+        targetSdkVersion 28
+        versionCode 1
+        versionName "1.0"
```
Comment (xdevl) [inline SimulatorApp/build.gradle]: I believe we should be using a 3 digits versioning scheme here

File: bintray.gradle [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+        licenses = ['Apache-2.0']
+        labels = ['payment', 'android']
+        websiteUrl = siteUrl
+        issueTrackerUrl = 'https://github.com/Aevi-UK/enabled-device-api/issues'
+        vcsUrl = gitUrl
+        publicDownloadNumbers = true
+        version {
+            name = "$project.apiVersion"
+            desc = "Aevi DMS Client"
+            vcsTag = "$project.apiVersion"
```
Comment (xdevl) [inline bintray.gradle]: Aren't we meant to set this property to the checked out git commit hash?

Comment (slambang) [reply bintray.gradle]: I originally pushed a commit to publish the library as an AAR, but we decided to go with JAR instead. The "version" property you see above is leftover form the AAR publication and is no longer needed, so I will remove it.

**Pull request merged**